staging/src/k8s.io/client-go/tools/portforward/portforward.go

```diff
@@ -110,8 +111,68 @@ func parsePorts(ports []string) ([]ForwardedPort, error) {
 	return forwards, nil
 }
 
+type listenAddress struct {
+	address  string
+	protocol string
+}
+
+func parseAddresses(addressesToParse []string) ([]listenAddress, error) {
+	var addresses, parsed []listenAddress
+	var al []string
+	// replace non IP entries with their resolved addresses or remove completely
+	// "localhost" will get replaced with 127.0.0.1 and ::1
+	for _, addr := range addressesToParse {
+		if addr == "localhost" {
+			al = append(al, "127.0.0.1")
+			al = append(al, "::1")
+		} else if net.ParseIP(addr) != nil {
+			al = append(al, addr)
+		} else {
+			var hosts []string
+			hosts, err := net.LookupHost(addr)
+			if err != nil {
+				return nil, fmt.Errorf("%s can not be parsed into valid listen address", addr)
+			}
+			al = append(al, hosts...)
+		}
+	}
+	addressesToParse = al
+	// assemble and return []listenAddress to listen on
+	for _, address := range addressesToParse {
+		if net.ParseIP(address).To4() != nil {
+			parsed = []listenAddress{{address: address, protocol: "tcp4"}}
```

Comment:
It seems `parsed` will always have one item, why make it a list?

Reply:
you're completely right, leftover from some refactoring